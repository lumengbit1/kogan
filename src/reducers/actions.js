import {FETCHDATA_ACTION} from './constants'

export const initListAction=(jsondata)=>({
    type:FETCHDATA_ACTION,
    data:jsondata
});


export const getEventList=()=>{
    return (dispatch)=>{
         return fetch('http://wp8m3he1wt.s3-website-ap-southeast-2.amazonaws.com/api/products/1')
            .then((res)=>{
                return res.json();       
            }).then((json)=>{
                dispatch(initListAction(json))
            })
            
 
    }
};

