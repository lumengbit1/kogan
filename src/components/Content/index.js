import React, { Component } from "react";
import './../../css/style.css';
import {connect} from 'react-redux';
import {getEventList} from './../../reducers/actions'


class Content extends Component{


    componentDidMount(){
        const {onfetchData} = this.props;
        onfetchData();
    }

    render(){
        const {content} = this.props;
        return(

            <div className='content'>
                {content!=null?
                    content.map((item)=>{
                        return(
                            <div>
                                <div>
                                    <h1>{item.title}</h1>
                                    <p>{item.description}</p>
                                </div>
                            </div>
                        )
                    }) :''
                }                
            </div>
        )
    }
}

// Map Redux state to component props
export const mapStateToProps=(state)=> {
    return {
        content: state.fetchdata
    }
}

// Map Redux actions to component props
export const mapDispatchToProps=(dispatch)=> {
    return {
        onfetchData: () => dispatch(getEventList())
    }
}

const IContent = connect(
    mapStateToProps,
    mapDispatchToProps
  )(Content)

export default IContent;