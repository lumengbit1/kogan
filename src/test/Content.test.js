import React from 'react'
import Enzyme,{ configure, shallow,mount,render } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Content from './../components/Content'
import configureMockStore from 'redux-mock-store'
import { Provider} from 'react-redux'
import sinon from 'sinon'
import {mapStateToProps,mapDispatchToProps} from './../components/Content'
import {getProducts} from "./../Reducers/actions"
import thunk from 'redux-thunk'
import * as ContentFunc from './../components/Content'


Enzyme.configure({ adapter: new Adapter() });

function setup() {
    const props = {
        calcWeight: jest.fn()
    }
    const middlewares = [thunk];
    const mockStore = configureMockStore(middlewares);
    let store=mockStore();
    const enzymeWrapper = mount(
        <Provider store={store} >
            <Content {...props}/>
        </Provider>)
  
    return {
      props,
      enzymeWrapper
    }
  }

describe('mapStateToProps,mapDispatchToProps Testing',()=>{
    let wrapper, store;
    const middlewares = [thunk];
    const mockStore = configureMockStore(middlewares);
    beforeEach(()=>{
        const initialState = {
            content: 1,
        };
        store=mockStore(initialState);
        wrapper = mount(
            <Provider store={store} >
                <Content />
            </Provider>)
    })

    it('onfetchData Testing',()=>{

        const dispatchSpy = sinon.spy();
        const {onfetchData} = mapDispatchToProps(dispatchSpy);
        onfetchData();
        const expectedAction = getProducts();
        const spyLastCall = dispatchSpy.args[0][0];
        expect(spyLastCall.types).toEqual(expectedAction.types);

    })
})


describe('render snapshop testing',()=>{
    it('renders correctly', () => {
        let wrapper, store;
        const middlewares = [thunk];
        const mockStore = configureMockStore(middlewares);
        store=mockStore();
        wrapper = render(
            <Provider store={store} >
                <Content />
            </Provider>);
        
                                     
        expect(wrapper).toMatchSnapshot();
      });
})



