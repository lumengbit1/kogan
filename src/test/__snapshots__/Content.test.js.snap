// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render snapshop testing renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "content",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "for": "average_cubic_weight",
        },
        "children": Array [
          Object {
            "data": "average cubic weight:",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "label",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "style": "display:inline",
          },
          "children": Array [
            Object {
              "data": "0kg",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "p",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {},
            "children": Array [],
            "name": "br",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {},
              "children": Array [
                Object {
                  "data": "Calculate Weight",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "button",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "for": undefined,
        },
        "x-attribsPrefix": Object {
          "for": undefined,
        },
      },
      Object {
        "attribs": Object {
          "style": "display:inline",
        },
        "children": Array [
          Object {
            "data": "0kg",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "p",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {},
          "children": Array [],
          "name": "br",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {},
            "children": Array [
              Object {
                "data": "Calculate Weight",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {},
            "x-attribsPrefix": Object {},
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "for": "average_cubic_weight",
          },
          "children": Array [
            Object {
              "data": "average cubic weight:",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "label",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "for": undefined,
          },
          "x-attribsPrefix": Object {
            "for": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "style": undefined,
        },
      },
      Object {
        "attribs": Object {},
        "children": Array [],
        "name": "br",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "data": "Calculate Weight",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "style": "display:inline",
          },
          "children": Array [
            Object {
              "data": "0kg",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "p",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "for": "average_cubic_weight",
            },
            "children": Array [
              Object {
                "data": "average cubic weight:",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "label",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "for": undefined,
            },
            "x-attribsPrefix": Object {
              "for": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "data": "Calculate Weight",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {},
          "children": Array [],
          "name": "br",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "style": "display:inline",
            },
            "children": Array [
              Object {
                "data": "0kg",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "p",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "for": "average_cubic_weight",
              },
              "children": Array [
                Object {
                  "data": "average cubic weight:",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "label",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "for": undefined,
              },
              "x-attribsPrefix": Object {
                "for": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "style": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
